File: Java/DesignPatterns/src/test/java/designPatterns/creational/abstractFactory/MainAbstractFactoryTest.java
package designPatterns.creational.abstractFactory;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class MainAbstractFactoryTest {
  private AbstractFactory factory;
  private AbstractProduct.A productA;
  private AbstractProduct.B productB;

  @BeforeEach
  void setUp() {
    factory = new ConcreteFactory();
    productA = factory.createProductA();
    productB = factory.createProductB();
  }

  @Test
  void equalsTest() {
    assertThat(productA.displayA()).isEqualTo("Product A");
    assertThat(productB.displayB()).isEqualTo("Product B");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/creational/abstractFactory/MainAbstractFactoryTest.kt
package creational.abstractFactory

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainAbstractFactoryTest {

  private lateinit var factory: AbstractFactory
  private lateinit var productA: AbstractProduct.A
  private lateinit var productB: AbstractProduct.B

  @BeforeEach
  internal fun setUp() {
    factory = ConcreteFactory()
    productA = factory.createProductA()
    productB = factory.createProductB()
  }

  @Test
  internal fun equalsTest() {
    assertThat(productA.displayA()).isEqualTo("Product A")
    assertThat(productB.displayB()).isEqualTo("Product B")
  }
}